src/ert/config/gen_kw_config.py

```diff
@@ -58,18 +59,44 @@ def _get_abs_path(file: Optional[str]) -> Optional[str]:
     return file
 
 
+@dataclass
+class TransformFunctionDefinition:
+    func_name: str
+    param_func_name: str
+    values: List[Any]
+
+
 @dataclass
 class GenKwConfig(ParameterConfig):
     template_file: Optional[str]
     output_file: Optional[str]
-    transfer_function_definitions: List[str]
+    transfer_function_definitions: List[TransformFunctionDefinition]
     forward_init_file: Optional[str] = None
 
     def __post_init__(self) -> None:
         self.transfer_functions: List[TransferFunction] = []
         for e in self.transfer_function_definitions:
-            self.transfer_functions.append(self._parse_transfer_function(e))
-
+            if isinstance(e, str):
```

Comment:
Given that we have to check whether `transfer_function_definitions` is a `str` or `dict`, does it make sense to type it as `List[TransformFunctionDefinition]`?

Reply:
I will add a Union